hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/ipc/TestProtoBufRpc.java

```diff
@@ -329,7 +331,30 @@ public void testProtoBufRandomException() throws Exception {
           .isEqualTo(RpcErrorCodeProto.ERROR_APPLICATION);
     }
   }
-  
+
+  @Test (timeout=5000)
+  public void testProtoBufReconstructableException() throws Exception {
+    //No test with legacy
+    assumeFalse(testWithLegacy);
+    TestRpcService client = getClient(addr, conf);
+
+    try {
+      client.error3(null, newEmptyRequest());
+    } catch (ServiceException se) {
+      assertThat(se.getCause()).isInstanceOf(RemoteException.class);
```

Comment:
if you use the AssertJ assertThat() (which a lot of new code does) you can explictily use an .isInstanceOf() assert and get more detail on the error

Reply:
Updated the following assert, please have a look.